src/components/Form/PasswordWidget.tsx

```diff
@@ -0,0 +1,126 @@
+import {
+	Box,
+	FormControl,
+	IconButton,
+	InputAdornment,
+	TextField,
+	Typography,
+} from '@mui/material';
+import * as React from 'react';
+import { Visibility, VisibilityOff } from '@mui/icons-material';
+import { WidgetProps } from '@rjsf/utils';
+import zxcvbn from 'zxcvbn';
+
+const STRENGTH_TITLES = ['Very weak', 'Weak', 'Good', 'Strong', 'Very strong'];
+const STRENGTH_STYLES = [
+	{ width: 0 },
+	{ width: '25%', backgroundColor: 'orange' },
+	{ width: '50%', backgroundColor: 'yellow' },
+	{ width: '75%', backgroundColor: 'green' },
+	{ width: '100%', backgroundColor: 'darkgreen' },
+];
+
+export interface PasswordStrengthProps {
+	password?: string;
+}
+
+const PasswordStrength = ({ password }: PasswordStrengthProps) => {
+	const [strengthScore, setStrengthScore] = React.useState<
+		number | undefined
+	>();
+
+	React.useEffect(() => {
+		if (password) {
+			try {
+				const { score } = zxcvbn(password);
+				setStrengthScore(score);
+			} catch {
+				// Ignore any errors, as we only want to show the strength meter if it is available.
+			}
+		}
+	}, [password]);
+
+	if (!password || strengthScore === undefined) {
+		return null;
+	}
+
+	return (
+		<Box>
+			<Typography>
+				Password strength: <em>{STRENGTH_TITLES[strengthScore]}</em>
+			</Typography>
+			<Box
+				sx={[
+					STRENGTH_STYLES[strengthScore],
+					{ height: '4px', transition: 'all ease-out 150ms' },
+				]}
+			/>
+		</Box>
+	);
+};
+
+export const PasswordWidget = ({
```

Comment:
nit: Might be better organized if we move the Widgets into their own folder `/src/components/Form/Widgets`, what do you think? I know we just have two right now but I plan on bringing over the FileWidget that Otavio has been working on after this PR is complete, and I won't be surprised if we find more widgets that we want to exist in this library